hw/xbox/nv2a/pgraph_texture_replacer.c

```diff
@@ -0,0 +1,26 @@
+#include "pgraph_texture_replacer.h"
+
+
+void texture_download(TextureBinding *texture, SurfaceBinding *surface)
+{
+    void *pixels = surface->texture;
+     char filename[20];
+
+    snprintf(filename, sizeof(filename), "%lu.bmp", surface->vram_addr);
+   
+    SDL_Surface * surf = SDL_CreateRGBSurfaceFrom(pixels, surface->width, surface->height, 8*4, surface->width*4, 0,0,0,0);
```

Comment:
I don't think xemu makes any guarantees that the surface format will be converted to RGBA, so you probably need to check the surface's format or BPP to decide on the appropriate size and may need to ask SDL to convert stuff for you.